sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -455,16 +458,24 @@ protected final void removeTemporaryFiles(List<String> knownFiles, PipelineOptio
       // (produced by successfully completed bundles).
       // This may still fail to remove temporary outputs of some failed bundles, but at least
       // the common case (where all bundles succeed) is guaranteed to be fully addressed.
-      Collection<String> matches = factory.match(factory.resolve(tempDirectory, "*"));
```

Comment:
This is a workaround because IOChannelFactory.remove() doesn't support recursively delete yet.
Ideally once we have that, this code could be simplified as factory.remove(tempDirectory).

Maybe take a look at https://issues.apache.org/jira/browse/BEAM-1041
I am happy to work with you, if you would like to help us to address BEAM-1041.